### ISSUE TITLE
Support minOccurs=0

### ISSUE THREAD

**Issue by pauldraper** [TimLuq/wsdl-to-ts#8]
Should be optional or undefined if minOccurs=0

**Comment by TimLuq** [comment]
Unfortunately I don't believe this is something exposed by the soap library. If it were it could be added, otherwise this lib would need a complete rewrite with a custom XSD parser.

**Comment by pauldraper** [comment]
I see that now :(

As is, the wsdl-to-ts cannot really be used with TS's strict nullable setting, as all elements become required.

**Comment by pauldraper** [comment]
(Thanks for the quick response, BTW! Great project.)

**Comment by TimLuq** [comment]
Yes, the way to handle the situation currently is to either generate the types once and then do manual corrections or, it pains me to say, have objects containing properties whose `undefined` or `null` values has been cast to `any`.

```ts
interface ISomeInterface {
  description: string; // : string | null - in reality supports xsi:nil="true"
  name: string;
  restricted: boolean; // ?: boolean - in reality minOccurs="0"
}
const o: ISomeInterface = {
  description: null as any,
  name: "example",
  restricted: undefined as any,
};
```
